### PR TITLE
COST-1196: Create work type specific workers.

### PR DESCRIPTION
**Pull request by myersCody**
## Testing with multiworker docker compose file

You will need to bring up the local docker using the multiworker. The easiest way I have found to do this is with:
```
(koku) bash-3.2$ export compose_file=testing/compose_files/docker-compose-multiworker.yml
(koku) bash-3.2$ printenv | grep compose
compose_file=testing/compose_files/docker-compose-multiworker.yml
(koku) bash-3.2$ make docker-up
/usr/local/bin/docker-compose -f testing/compose_files/docker-compose-multiworker.yml up --build -d
```
The `compose_file` var is then picked up by the makefile and will trigger for all our docker-compose make commands. Then when you are done, you can run `(koku) bash-3.2$ unset compose_file` to remove it from the env and go back to normal. 

#### Step One:  Winter is coming: aka pull up the logs
I recommend bringing up a docker compose  logs for each worker you want to monitor. For me it makes it easier to see when each task is triggered and to make sure each task was triggered.
```
docker-compose -f testing/compose_files/docker-compose-multiworker.yml logs -f download-worker
docker-compose -f testing/compose_files/docker-compose-multiworker.yml logs -f summary-worker
docker-compose -f testing/compose_files/docker-compose-multiworker.yml logs -f cost-model-worker
docker-compose -f testing/compose_files/docker-compose-multiworker.yml logs -f refresh-worker
```
#### Step Two: Upload a Provider
I like to start with a single provider because it was easier to track then our normal dummy data upload. For my example, I used gcp because it is essentially muscle memory for me to create a GCP source at this point.
```
gcp-source gcp_name=CodyBiggoStormsMyers
```
Next, we will have to watch the downloader, summary, and refresh worker to see that the celery tasks are picked up. 

###### Clarification ######
Knowing when cost-model worker should be triggered is a little tricky. Anything through the `cost_model` endpoint, actually goes to the priority-worker because it is a user initiated task so we want it to be a high priority task. Therefore, the cost_model worker actually pick up the cost_model updates for when a new report is processed.

--------

The best way to see the cost_model updates for when a new report is processed is to use our `make load-test-customer-data` command. 

e2e Changes: https://github.com/RedHatInsights/e2e-deploy/pull/2951
--------
*Smoke Tests*
```
=============================================================== short test summary info ================================================================
FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test_cost_model.py::test_api_cost_model_recalc_window - AssertionError: as...
FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test_metrics.py::test_api_ocp_metrics - AssertionError: Expected metrics n...
FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test_user.py::test_api_settings_ocp_tag_enablement - TypeError: argument o...
======================================= 3 failed, 3954 passed, 80 skipped, 1736 deselected in 1363.63s (0:22:43) =======================================
```
I believe the failures are unrelated to my changes.

### CODE REVIEW

File: testing/compose_files/docker-compose-multiworker.yml [hunk 1]
```diff
@@ -219,7 +219,7 @@ services:
       hostname: koku-worker-1-fsfsgr
       image: koku_base
       working_dir: /koku/koku
-      entrypoint: ['watchmedo', 'auto-restart', '--directory=./', '--pattern=*.py', '--recursive', '--', 'celery', '-A', 'koku', 'worker', '-l', 'info', '-Q', 'celery,download,remove_expired,reporting,process,upload,customer_data_sync,delete_archived_data,query_upload']
+      entrypoint: ['watchmedo', 'auto-restart', '--directory=./', '--pattern=*.py', '--recursive', '--', 'celery', '-A', 'koku', 'worker', '-l', 'info', '-Q', 'celery,remove_expired,reporting,process,upload,customer_data_sync,delete_archived_data,query_upload']
```
Comment (maskarb) [inline testing/compose_files/docker-compose-multiworker.yml:222]: This file no longer needs 10 workers looking at all the queues. We can probably create 1 worker per queue instead

File: koku/masu/processor/orchestrator.py [hunk 2]
```diff
@@ -184,7 +185,7 @@ def start_manifest_processing(
                     provider_uuid,
                     report_month,
                     report_context,
-                )
+                ).set(queue=GET_REPORT_FILES_QUEUE)
             )
             LOG.info("Download queued - schema_name: %s.", schema_name)
```
[the remark this answers is not in the document]
Comment (myersCody) [reply koku/masu/processor/orchestrator.py:191]: This is a good point @maskarb I know that I needed to set the queue above to get the downloader to switch over to the right worker. I should double check the worker logs to see if the sumarize_reports is running int he right queue too. 

Double checking that now!

File: testing/compose_files/docker-compose-multiworker.yml (hunk 1 above)
Comment (myersCody) [reply testing/compose_files/docker-compose-multiworker.yml:222]: Yeah, I will see what else we can trim down from this list.

File: koku/masu/processor/orchestrator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (myersCody) [reply koku/masu/processor/orchestrator.py:193]: So, it turns out that I need to set both of these or they will default to the celery queue and be picked up by the koku-worker.

**Review by adberglund — APPROVED** [review]

**Pull request merged**